### DISCUSSION TITLE
Alignment for ASR Models

### DISCUSSION THREAD

**Discussion by ahkarami** [NVIDIA-NeMo/NeMo#4132] [Q&A] [answered]
Hi,
Does NeMo ASR models support `alignment`? I mean how one can find that each word (or token) extract from which timestamp?
Best

**Comment by titu1994** [comment, marked as answer]
Not yet, but we are working on it. Realistically speaking it might take a month or two to have a prototype, and a month after that for docs / tutorial. 

CC @erastorgueva-nv @GNroy

**Reply by titu1994** [reply]
If you're ok with non forced alignment, RNNT has alignments during inference. Refer to the RNNT tutorials.

**Reply by ahkarami** [reply]
Dear @titu1994 
Thanks a lot for your great answer
Best

**Reply by GNroy** [reply]
@ahkarami If you have a CTC model, you can do forced alignment with https://github.com/NVIDIA/NeMo/blob/main/examples/asr/experimental/k2/align_speech_parallel.py (requires [k2](https://github.com/k2-fsa/k2) installed)
This script takes a model and a mainfest and produces CTMs for the data. In the current state it supports forced and argmax (decoding) alignments for CTC models. The RNNT forced alignment support is in progress (you can find a prototype in https://github.com/NVIDIA/NeMo/pull/3962)

**Reply by ahkarami** [reply]
Thanks a lot @GNroy
